File: 0001.两数之和.cpp
#include <unordered_map>
#include <vector>

class Solution {
 public:
  std::vector<int> twoSum(std::vector<int>& nums, int target) {
    std::unordered_map<int, int> existedNumbers;  // value index
    for (int i = 0; i < nums.size(); i++) {
      if (existedNumbers.count(target - nums[i]) == 0) {
        existedNumbers.insert({nums[i], i});
      } else {
        return {i, existedNumbers[target - nums[i]]};
      }
    }
    return {};
  }
};

File: 0001.两数之和.go
package leetcode

func twoSum(nums []int, target int) []int {
	existedNumbers := map[int]int{} // value index
	for currentIndex, currentNumber := range nums {
		if targetIndex, ok := existedNumbers[target-currentNumber]; ok {
			return []int{targetIndex, currentIndex}
		}
		existedNumbers[currentNumber] = currentIndex
	}
	return nil
}
